arrow/arrow.py

```diff
@@ -563,7 +573,11 @@ def span(
             floor = self.__class__(*values, tzinfo=self.tzinfo)  # type: ignore
 
             if frame_absolute == "week":
-                floor = floor.shift(days=-(self.isoweekday() - 1))
+                delta = 0
+                # if week_start is greater than self.isoweekday() go back one week by setting delta = 7
+                if week_start > self.isoweekday():
+                    delta = 7
```

Comment:
This could probably be made into a clean 1 liner: `delta = 7 if week_start > self.isoweekday() else 0` and include the comment on top of it :).

Reply:
Thinking about it though... since we now have the `not 1 <= week_start <= 7` check above, isn't it never going to be the case that week_start is greater than isoweekday?

Reply:
`isoweekday` refers to the `arrow` object, which might be 3 (wednesday). Assuming the user wants to span from last Thursday, `week_start` must be set to 4. This would span the week from last Thursday to coming Thursday.

Reply:
Updated the code and made a one liner.